src/functional/program_processors/codegens/gtfn/gtfn_module.py

```diff
@@ -48,25 +52,87 @@ def __call__(
+                )  # TODO std::forward, type and max_neighbors)
+            elif isinstance(connectivity, Dimension):
+                pass
+            else:
+                raise ValueError(
+                    f"Expected offset provider `{name}` to be a "
+                    f"`Connectivity` or `Dimension`, but got "
+                    f"{type(connectivity).__name__}"
+                )
+        rendered_connectivity_args = ", ".join(connectivity_args)
+
+        # TODO(tehrengruber): The backend expects all arguments to a stencil closure to be a SID
+        #  so transform all scalar arguments that are used in a closure into one before we pass
+        #  them to the generated source. This is not a very clean solution and will fail when
+        #  the respective parameter is used elsewhere, e.g. in a domain construction, as it is
+        #  expected to be scalar there (instead of a SID). We could solve this by:
+        #   1.) Extending the backend to support scalar arguments in a closure (as in embedded
+        #       backend).
+        #   2.) Use SIDs for all arguments and deref when a scalar is required.
+        closure_scalar_parameters = (
+            eve.trees.pre_walk_values(
+                eve.utils.XIterable(program.closures).getattr("inputs").to_list()
+            )
+            .if_isinstance(itir.SymRef)
+            .getattr("id")
+            .map(str)
+            .to_list()
         )
-        function = interface.Function(program.id, parameters)
 
-        rendered_params = ", ".join(
-            ["gridtools::fn::backend::naive{}", *(p.name for p in parameters)]
-        )
-        decl_body = f"return generated::{function.name}()({rendered_params});"
+        parameter_args = ["gridtools::fn::backend::naive{}"]
+        for p in parameters:
+            if isinstance(p, (interface.ScalarParameter, interface.BufferParameter)):
+                if isinstance(p, interface.ScalarParameter) and p.name in closure_scalar_parameters:
+                    parameter_args.append(f"gridtools::stencil::global_parameter({p.name})")
+                else:
+                    parameter_args.append(p.name)
+
+        rendered_parameter_args = ", ".join(parameter_args)
+        decl_body = f"return generated::{function.name}({rendered_connectivity_args})({rendered_parameter_args});"
         decl_src = cpp_interface.render_function_declaration(function, body=decl_body)
         stencil_src = gtfn_backend.generate(program, **inp.kwargs)
         source_code = interface.format_source(
             self.language_settings,
             f"""
-            #include <gridtools/fn/backend/naive.hpp>
-            {stencil_src}
-            {decl_src}
-            """.strip(),
+                    #include <gridtools/fn/backend/naive.hpp>
+                    #include <gridtools/stencil/global_parameter.hpp>
+                    {stencil_src}
+                    using gridtools::fn::sid_neighbor_table::as_neighbor_table;
```

Comment:
```suggestion
```
and use the full name above or create a namespace alias here, if you think it's too long and ugly.